src/xrpld/overlay/detail/PeerImp.cpp

```diff
@@ -3008,7 +3243,41 @@ PeerImp::getLedger(std::shared_ptr<protocol::TMGetLedger> const& m)
                 }
 
                 JLOG(p_journal_.trace())
-                    << "getLedger: Failed to find peer to relay request";
+                    << "getLedger: Don't have ledger with hash " << ledgerHash
+                    << ": " << mHash;
+
+                if (m->has_querytype() && !m->has_requestcookie())
+                {
+                    // Attempt to relay the request to a peer
+                    // Note repeated messages will not relay to the same peer
+                    // before `getLedgerInterval` seconds. This prevents one
+                    // peer from getting flooded, and distributes the request
+                    // load. If a request has been relayed to all eligible
+                    // peers, then this message will not be relayed.
+                    if (auto const peer = getPeerWithLedger(
+                            overlay_,
+                            ledgerHash,
+                            m->has_ledgerseq() ? m->ledgerseq() : 0,
+                            this,
+                            [&](Peer::id_t id) {
+                                return app_.getHashRouter()
+                                    .shouldProcessForPeer(
+                                        mHash, id, getledgerInterval);
+                            }))
+                    {
+                        m->set_requestcookie(id());
+                        peer->send(std::make_shared<Message>(
+                            *m, protocol::mtGET_LEDGER));
+                        JLOG(p_journal_.debug())
+                            << "getLedger: Request relayed to peer ["
+                            << peer->id() << "]: " << mHash;
+                        return ledger;
+                    }
+
+                    JLOG(p_journal_.trace())
```

Comment:
[nit]: Should this be a `warn` instead of `trace`? Not having a peer to relay the request may indicate some configuration or environment issues.

Reply:
Not necessarily, though. It could mean that the node has already sent the request to all it's peers. Also not that the original message is `trace`.

But there's something odd here. It looks like this code block was somehow duplicated! I must have messed up resolving a conflict when I rebased from `master` to `develop`. I've removed the duplicate.